src/main/kotlin/org/rust/ide/annotator/fixes/ChangeFunctionSignatureFix.kt

```diff
@@ -0,0 +1,200 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.annotator.fixes
+
+import com.intellij.codeInspection.LocalQuickFixAndIntentionActionOnPsiElement
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import org.rust.cargo.project.workspace.PackageOrigin
+import org.rust.ide.annotator.getFunctionCallContext
+import org.rust.ide.presentation.render
+import org.rust.ide.presentation.renderInsertionSafe
+import org.rust.ide.refactoring.changeSignature.Parameter
+import org.rust.ide.refactoring.changeSignature.ParameterType
+import org.rust.ide.refactoring.changeSignature.RsChangeFunctionSignatureConfig
+import org.rust.ide.refactoring.changeSignature.runChangeSignatureRefactoring
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.containingCrate
+import org.rust.lang.core.psi.ext.isMethod
+import org.rust.lang.core.psi.ext.valueParameters
+import org.rust.lang.core.types.implLookup
+import org.rust.lang.core.types.ty.TyUnknown
+import org.rust.lang.core.types.type
+
+/**
+ * Requires that the number of arguments is larger than the number of parameters.
+ */
+class ChangeFunctionSignatureFix private constructor(
+    arguments: RsValueArgumentList,
+    function: RsFunction,
+    private val direction: ArgumentScanDirection
+) : LocalQuickFixAndIntentionActionOnPsiElement(arguments) {
+    private val fixText = run {
+        val signature = calculateSignature(function, arguments, direction)
+        val argumentInsertions = signature.withIndex().filter { (_, item) -> item is SignatureMember.Argument }
```

Comment:
```suggestion
        val argumentInsertions = signature.filterIsInstance<SignatureMember.Argument>().withIndex()
```

Reply:
This actually broke the logic, since the indices need to be collected before the filtering :sweat_smile:.